istioctl/cmd/istioctl/rbac/cmd.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2018 Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rbac
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+)
+
+type subjectArgs struct {
+	user       string
+	groups     string
+	properties []string
+}
+
+type actionArgs struct {
+	namespace  string
+	service    string
+	method     string
+	path       string
+	properties []string
+}
+
+// cani() allows user to query Istio RBAC effect for a specific request.
+func cani() *cobra.Command {
+	subject := subjectArgs{}
+	action := actionArgs{}
+
+	cmd := &cobra.Command{
+		Use:   "can-i",
+		Short: "Query Istio RBAC policy effect for a specific request",
+		Long: `
+This command lets you query whether a specific request is allowed or denied under current Istio RBAC
+policies. You can construct a fake request with custom subject and target to check if your Istio RBAC
+policies is working as expected. Note the fake request is only used locally to evaluate the effect of
+the Istio RBAC policies, no actual request will be issued.
+`,
+		Example: `# Query if user test is allowed to GET pageA of rating service in default namespace.
+istioctl experimental rbac can-i --subject-user test --target-method=GET --target-service=rating --target-path=pageA --target-namespace=default`,
```

Comment:
IMHO this is a bit too verbose. What about something like...

`istioctl experimental rbac can <user|service> <method> <service> <path> [-n namespace]`
`istioctl experimental rbac can test-user GET product-page /v1/health [-n namespace]`

Reply:
Done.